test/test_xla_sharding.py

```diff
@@ -18,24 +19,17 @@
                 )
 class XlaShardingTest(unittest.TestCase):
 
-  @unittest.skip("work-in-progress")
   def test_xla_sharded_tensor(self):
-    # TODO(244003536) re-enable when new test cases are ready.
     n_devices = xm.xrt_world_size()
     mesh_shape = (1, n_devices)
-    partition_spec = (1,)
-    t1 = torch.tensor([2.0, 3.0], dtype=torch.float, device=xm.xla_device())
-    t1_sharded = XLAShardedTensor(t1, mesh_shape, partition_spec)
-    t2 = torch.tensor([2.0, 3.0], dtype=torch.float, device=xm.xla_device())
-    t3 = torch.add(t1_sharded, t2)
+    partition_spec = (0, 1)
+    xt1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                       dtype=torch.float,
+                       device=xm.xla_device())
+    xst1 = xs.mark_sharding(xt1, mesh_shape, partition_spec)
```

Comment:
so xt1 stays as `torch.Tensor` but under the hood sharding spec is already being inserted? From here on, should people use `xt1` or `xst1`? This part is a bit confusing.

Reply:
Yes, annotation doesn't change the type of tensor. Annotation is attached to the backing IR. people can use both `xt1` and `xst1` because `xst1` is just a wrapper that dispatches to `xt1` and it's XLA backend. This reason for this is that the user can simply take their existing code and don't change/replace tensors with sharded tensors explicitly (so the true annotating); also returned handle to the sharded tensor provides an access to XLAShardedTensor specific APIs (e.g., sharding_spec, accessing shards, etc.).

Reply:
should we consider rename `mark_sharding` to `mark_sharding_` since it is actully an inplace op?

an op that both inplace update the op and returns it is a bit counterintuitive. IMO we should at least make it clear that it is inplace.